### PR TITLE
new tutorial

### PR DESCRIPTION
**Pull request by PhilipVinc**

### CODE REVIEW

**Comment by PhilipVinc** [comment]
Yeah, of course. (Do you know his GitHub handle?)
I wanted to finish it before, though.

My plan was to show how to implement by hand the expectation values and gradient of energy...

**Comment by gcarleo** [comment]
@PhilipVinc it's jrm874

File: docs/tutorials/variational_intro.ipynb [hunk 1]
```diff
@@ -0,0 +1,1625 @@
...
...2fRvzftM7kaoGMQigAArTLl5CRJ0mebCk2uBOgYhCIAQKucnZ4gu82q7fsrtK2ozOxygHZHKAIAtEpkSLAmDIyTxNkidE2EIgBAq00dWt+F9m9CEbogQhEAoNV+NiRRNqtF3+1zaffBCrPLAdoVoQgA0Go9wu3KTIuVJC3cyFVo6FoIRQCANjl/RIok6aPsApMrAdoXoQgA0CZThibJbrMqp6hMWwuZyBFdB6EIANAmztBg30SO/+JsEboQQhEAoM3OH9FTUn0XGvdCQ1cRcKFo165dmjVrlvr166fQ0FD1799f8+bNU01NTYvbTZw4URaLpcly4403dlLVANC1TBqSoHC7TfklVVqfd9jscoB2EWR2AW21detWeb1evfjiixowYIA2bdqk6667ThUVFfrzn//c4rbXXXedHnzwQd/rsLCwji4XALqkkGCbpgxN0vvr8/XP9fka1SfG7JKAnyzgQtHUqVM1depU3+u0tDTl5OTo+eefP24oCgsLU1JSUkeXCADdwi9O7aX31+fr4+wC3XfeSQq128wuCfhJAq77rDmlpaWKiTn+/6W8+eabiouL09ChQzV37lxVVlYes63b7ZbL5WqyAAB+MDYtVr16hKrMXafPNjNnEQJfwIei3NxcPf3007rhhhtabHfllVfqjTfe0LJlyzR37ly9/vrruuqqq47Zfv78+XI6nb4lNTW1vUsHgIBmtVp0yaj6v41/X7vX5GqAn85i+MllA3fffbceeeSRFtts2bJF6enpvtf5+fk688wzNXHiRP31r39t089bunSpJk2apNzcXPXv3/+o991ut9xut++1y+VSamqqSktLFRUV1aafBQBd1d7DlTr90WUyDOmLO89S71jGasK/uFwuOZ3OVn1/+82Yojlz5mjmzJkttklLS/M9Lygo0FlnnaVx48bppZdeavPPy8jIkKRjhiKHwyGHw9HmzwWA7qRXjzBNGBCnL7cd0D+y9mj2OYPNLgk4YX4TiuLj4xUfH9+qtvn5+TrrrLM0atQovfLKK7Ja294LmJ2dLUlKTk5u87YAgB9cOjpVX247oHfX7dGtkwYq2BbwIzPQTQXcv9z8/HxNnDhRvXv31p///Gft379fhYWFKiwsbNImPT1da9askSRt375dDz30kLKysrRr1y599NFHmjFjhs444wwNHz7crF0BgC5hyslJiotwq
```
Comment (jrm874) [inline docs/tutorials/variational_intro.ipynb:990]: Here I made a conceptual mistake. This line should be replaced by:

"Consider the loss function $E_\\theta$ to be optimized with respect to the set of variational parameters $\\{\\theta\\}$. The goal is to find the change of parameters $\\delta \\theta$ such that $E_{\\theta + \\delta \\theta}$ is minimized. Taylor expanding $E_{\\theta + \\delta \\theta}$ to first order in $\\delta \\theta$: \n"

Comment (jrm874) [inline docs/tutorials/variational_intro.ipynb:993]: To keep correcting the mistake this line should be replaced by:

"E_{\\theta+\\delta\\theta} = E_\\theta + \\sum_k\\frac{\\partial E_\\theta}{\\partial\\theta_k}\\delta\\theta_k + \\mathcal{O}\\big((\\delta \\theta)^2\\big) \n "

Comment (jrm874) [inline docs/tutorials/variational_intro.ipynb:996]: To keep correcting the mistake this line should be replaced by:

'' Besides minimizing  $E_{\\theta+\\delta\\theta}$ with respect to $\\delta \\theta$, we also require that the distance travelled in parameter space to be a small value $\\epsilon$. Distance in parameter space can be measured by the Euclidean distance: \n''

Comment (jrm874) [inline docs/tutorials/variational_intro.ipynb:1002]: To keep correcting the mistake this line should be replaced by:

"Thus, we can combine the notion that we aim at minimizing $E_{\\theta + \\delta \\theta}$ and traveling distance  $d^2 = \epsilon$ to obtain the Lagrangian that we want to minimize with respect to the parameter variations $\\delta \\theta$ ( $\\lambda$ is the Lagrange Multiplier):\n"

Comment (jrm874) [inline docs/tutorials/variational_intro.ipynb:1005]: To keep correcting the mistake this line should be replaced by:

"L = E_{\\theta + \\delta \\theta} + \\lambda (d^2-\\epsilon)" = E_\\theta + \\sum_k\\left( \\frac{\\partial E_\\theta}{\\partial\\theta_k} \\delta\\theta_k + \\lambda ((\\delta\\theta_k)^2 -\\epsilon)\\right).\n

Comment (jrm874) [inline docs/tutorials/variational_intro.ipynb:1008]: To keep correcting the mistake this line should be replaced by:

"The minimum of that function with respect to $\\delta \\theta$ is reached when $\\frac{\\partial L}{\\partial (\\delta\\theta_m)} = 0$:\n"

Comment (jrm874) [inline docs/tutorials/variational_intro.ipynb:1190]: To keep correcting the mistake this line should be replaced by:

"L = E_{\\theta + \\delta \\theta} + \\lambda (d^2-\epsilon) = E_\\theta + \\sum_k \\frac{\\partial E_\\theta}{\\partial\\theta_k} \\delta\\theta_k + \\lambda \\left( \\sum_{kk'} g_{kk'}(\\theta) \\delta \\theta_k \\delta \\theta_{k'} -\epsilon \\right).\n"

Comment (jrm874) [inline docs/tutorials/variational_intro.ipynb:1193]: To keep correcting the mistake this line should be replaced by:

"The minimum of that function with respect to $\\delta \\theta$ is reached when $\\frac{\\partial L}{\\partial (\\delta\\theta_m)} = 0$:\n"

**Review by jrm874 — CHANGES_REQUESTED** [review]
The tutorial looks good to me! I am suggesting a couple of stylistic changes (mostly to make a bit more formal what I wrote for the lectures).

I also realized that I made a small conceptual mistake when deriving the update rule for the vanilla gradient descent. I provide line-by-line suggestions to address this mistake. (if the line-by-line commenting is too annoying to implement I can also send you the corrected text for the cell)

Let me know what you think :)

**Comment by gcarleo** [comment]
thanks a lot @jrm874 ! would you mind committing directly the changes to notebook? I think you should be already able to push to this PR? @PhilipVinc

**Comment by gcarleo** [comment]
also, do I remember correctly that we also had a tutorial on fermions? I think it would be nice if at some point we could add that too!

**Comment by jrm874** [comment]
I have implemented the fixes, but it doesn't allow me to commit directly. It says that I don't have write access to netket

**Comment by jrm874** [comment]
> also, do I remember correctly that we also had a tutorial on fermions? I think it would be nice if at some point we could add that too!

Yes, we did a brief tutorial about fermions for part of the class. Although I think it can be significantly improved. What did you have in mind? A very introductory-level tutorial where we discuss the commutation rules of creation/annihilation operators and the sign bookkeeping of Jordan-Wigner mapping? I could write something along those lines with some examples on spin-polarized fermions and Hubbard model

**Comment by PhilipVinc** [comment]
@jrm874 you should be able to write now.

**Comment by jrm874** [comment]
Thanks @PhilipVinc! Just pushed the commit.
